### ISSUE TITLE
Feature Request: Integrate a fully functional Dark/Light Mode toggle icon.

### ISSUE THREAD

**Issue by kumartushar0605** [PorePranav/CampusUnify#6]
To enhance the user experience by integrating a Dark/Light Mode toggle icon that allows users to switch between themes seamlessly. This feature is essential for improving accessibility, reducing eye strain, and catering to user preferences for different viewing environments.

Kindly assign me this issue under HacktoberFest 2024.

**Comment by PorePranav** [comment]
Hi @kumartushar0605,
Thank you for opening up the issue. You can start working on the new feature/ issue. Good luck!

**Comment by kumartushar0605** [comment]
Hello @PorePranav  I have created the pull request please accept it

**Comment by Adebesin-Cell** [comment]
Hi @PorePranav,

Thanks again for your detailed response! Just wanted to update you on my current progress:

- I've built a custom hook, `useTheme`, which adapts to the system’s theme and also allows users to switch between dark and light modes.
- The theme preference persists using cookies, ensuring the user’s choice is remembered across sessions even on page reload.
- Next, I'll be focusing on the layout, refining the colors to better fit the design, and handling any edge cases I might come across.

I’ll keep you posted as I make more progress. Let me know if there are any specific areas you'd like me to prioritize or focus on.

Thanks again!

**Comment by PorePranav** [comment]
Hey @Adebesin-Cell 
Great to hear about your progress! Seems like a quality PR is coming up, thanks for the update. Let me know if you need any inputs or help from my side.

**Comment by Adebesin-Cell** [comment]
> Hey @Adebesin-Cell Great to hear about your progress! Seems like a quality PR is coming up, thanks for the update. Let me know if you need any inputs or help from my side.

Hi @PorePranav,

Thanks for the encouraging words! 

I’m currently running into an issue with setting up my server environment. I tried using random values for the environment variables, but it seems like the server isn’t starting up properly. 

Here’s a screenshot of the error (it’s still uploading as I type this). Do you have any suggestions on how I can resolve this? I'd appreciate any guidance you can provide.

Thanks in advance!

<img width="1000" alt="Screenshot 2024-10-06 at 8 34 44 PM" src="https://github.com/user-attachments/assets/eba55e59-89d3-44d9-8847-330f266bd866">

**Comment by PorePranav** [comment]
> > Hey @Adebesin-Cell Great to hear about your progress! Seems like a quality PR is coming up, thanks for the update. Let me know if you need any inputs or help from my side.
> 
> Hi @PorePranav,
> 
> Thanks for the encouraging words!
> 
> I’m currently running into an issue with setting up my server environment. I tried using random values for the environment variables, but it seems like the server isn’t starting up properly.
> 
> Here’s a screenshot of the error (it’s still uploading as I type this). Do you have any suggestions on how I can resolve this? I'd appreciate any guidance you can provide.
> 
> Thanks in advance!
> <img alt="Screenshot 2024-10-06 at 8 34 44 PM" width="1000" src="https://private-user-images.githubusercontent.com/66301634/373975002-eba55e59-89d3-44d9-8847-330f266bd866.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjgyNDU5NzAsIm5iZiI6MTcyODI0NTY3MCwicGF0aCI6Ii82NjMwMTYzNC8zNzM5NzUwMDItZWJhNTVlNTktODlkMy00NGQ5LTg4NDctMzMwZjI2NmJkODY2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDA2VDIwMTQzMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTllYTM4ODhmYmZlYzMwMTA3YjNiYzNlYzIzYWQ5MzRkMzgxMGVmZjliODgxMmVkOTE0NTMzYTcxMmUwN2Q1NjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.crB5mcI5tyANRD0n1XnfKxWzBVVajeNTx0oMT5QGsCs">

RAZORPAY_ID_KEY=rzp_test
RAZORPAY_SECRET_KEY=ABCD
RAZORPAY_WEBHOOK_SECRET=ABCD

Try setting up this in your config.env

**Comment by Adebesin-Cell** [comment]
It works now, thank you so much @PorePranav

**Comment by Adebesin-Cell** [comment]
Hi @PorePranav,

Thanks again for your timely responses! I’ve run into another issue: I need access to some protected routes so I can modify their UIs and see the changes in real-time to properly design them.

Could you provide a step-by-step guide on how I can access these routes for development? It would be really helpful!

Thanks in advance for your help.

**Comment by PorePranav** [comment]
> Hi @PorePranav,
> 
> Thanks again for your timely responses! I’ve run into another issue: I need access to some protected routes so I can modify their UIs and see the changes in real-time for better design.
> 
> Could you provide a step-by-step guide on how to access these routes during development? It would be super helpful!
> 
> Also, I’ve updated the UIs for the ones I could access in real-time. You can check them out in the PR: [CampusUnify #17](https://github.com/PorePranav/CampusUnify/pull/17).
> 
> Thanks in advance for your help!

A protected route in the frontend is a route that would require a user to be logged in. So, you can do the following things to resolve this issue:
1. Setup a Mongo DB url at server's `config.env` (I think you may have already done this).
2. Once this is done, go to the signup page `/signup` from the client (frontend)
3. Fill up the form, make sure that you use a valid email id and password (more than 8 characters) and sign up to the application.
4. Once you are signed in, you can access all the protected routes.

Thank you! Let me know if you need any further help!

**Comment by Adebesin-Cell** [comment]
> > Hi @PorePranav,
> > Thanks again for your timely responses! I’ve run into another issue: I need access to some protected routes so I can modify their UIs and see the changes in real-time for better design.
> > Could you provide a step-by-step guide on how to access these routes during development? It would be super helpful!
> > Also, I’ve updated the UIs for the ones I could access in real-time. You can check them out in the PR: [CampusUnify #17](https://github.com/PorePranav/CampusUnify/pull/17).
> > Thanks in advance for your help!
> 
> A protected route in the frontend is a route that would require a user to be logged in. So, you can do the following things to resolve this issue:
> 
> 1. Setup a Mongo DB url at server's `config.env` (I think you may have already done this).
> 2. Once this is done, go to the signup page `/signup` from the client (frontend)
> 3. Fill up the form, make sure that you use a valid email id and password (more than 8 characters) and sign up to the application.
> 4. Once you are signed in, you can access all the protected routes.
> 
> Thank you! Let me know if you need any further help!


Noted, thank you! Just to clarify, does the MongoDB URL need to be a valid, working MongoDB instance, or can I use a dummy string for testing purposes?

Thanks again for your help!

**Comment by PorePranav** [comment]
> > > Hi @PorePranav,
> > > Thanks again for your timely responses! I’ve run into another issue: I need access to some protected routes so I can modify their UIs and see the changes in real-time for better design.
> > > Could you provide a step-by-step guide on how to access these routes during development? It would be super helpful!
> > > Also, I’ve updated the UIs for the ones I could access in real-time. You can check them out in the PR: [CampusUnify #17](https://github.com/PorePranav/CampusUnify/pull/17).
> > > Thanks in advance for your help!
> > 
> > 
> > A protected route in the frontend is a route that would require a user to be logged in. So, you can do the following things to resolve this issue:
> > 
> > 1. Setup a Mongo DB url at server's `config.env` (I think you may have already done this).
> > 2. Once this is done, go to the signup page `/signup` from the client (frontend)
> > 3. Fill up the form, make sure that you use a valid email id and password (more than 8 characters) and sign up to the application.
> > 4. Once you are signed in, you can access all the protected routes.
> > 
> > Thank you! Let me know if you need any further help!
> 
> Noted, thank you! Just to clarify, does the MongoDB URL need to be a valid, working MongoDB instance, or can I use a dummy string for testing purposes?
> 
> Thanks again for your help!

You would need a valid MongoDB instance and a URL, the backend does depend upon it and without that you won't be able to start the backend server. You can get one at https://www.mongodb.com/ or host one locally.

**Comment by Adebesin-Cell** [comment]
Thanks for your response, @PorePranav! I’ll check it out and try setting up a valid MongoDB instance. Apologies for the late reply! I’ll see if I can get everything working by tomorrow.

**Comment by Adebesin-Cell** [comment]
Hi @PorePranav, 

I’m trying it out today, but I keep encountering the following error: 

```
MongooseError: Operation `users.insertOne()` buffering timed out after 10000ms
```

It seems to be timing out when attempting to insert data into the database. Do you have any suggestions on what might be causing this?

Thanks again for your help!

**Comment by PorePranav** [comment]
> Hi @PorePranav,
> 
> I’m trying it out today, but I keep encountering the following error:
> 
> ```
> MongooseError: Operation `users.insertOne()` buffering timed out after 10000ms
> ```
> 
> It seems to be timing out when attempting to insert data into the database. Do you have any suggestions on what might be causing this?
> 
> Thanks again for your help!

Hey @Adebesin-Cell,
I guess you are having issues with your MongoDB instance, can we connect over on Discord so that I can assist you better? My Discord id is pranav.js

**Comment by Adebesin-Cell** [comment]
> > Hi @PorePranav,
> > I’m trying it out today, but I keep encountering the following error:
> > ```
> > MongooseError: Operation `users.insertOne()` buffering timed out after 10000ms
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > It seems to be timing out when attempting to insert data into the database. Do you have any suggestions on what might be causing this?
> > Thanks again for your help!
> 
> Hey @Adebesin-Cell, I guess you are having issues with your MongoDB instance, can we connect over on Discord so that I can assist you better? My Discord id is pranav.js

That would be great, Pranav! I’ll send you a DM on Discord right now.

**Comment by Adebesin-Cell** [comment]
<img width="719" alt="Screenshot 2024-10-10 at 7 47 38 AM" src="https://github.com/user-attachments/assets/a47c71e3-5ca0-4965-acbe-84aaff55c18d">


Hey @PorePranav, I tried adding you on Discord, but it says you're not accepting friend requests (screenshot attached). My Discord username is **tolulope_adebesin**. Could you add me instead?

Thanks!

**Comment by Adebesin-Cell** [comment]
Hey @PorePranav,

I’ve resolved the MongoDB issues and made some updates to the PR. Could you kindly check it out and provide your feedback? I’d love to work on any suggestions or improvements you may have. 

Thanks a lot!
